File: src/2149-Rearrange-Array-Elements-by-Sign/rearrange_array_elements_by_sign.py
from typing import List

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        positive: List[int] = []
        negative: List[int] = []
        output: List[int] = []
        
        for n in nums:
            if n >= 0:
                positive.append(n)
            else:
                negative.append(n)
        
        for i in range(len(positive)):
            output.append(positive[i])
            output.append(negative[i])
            
        return output

            
if __name__ == "__main__":
    nums: List[int] = [3,1,-2,-5,2,-4]
    
    solution = Solution()
    
    print(solution.rearrangeArray(nums))

File: src/2149-Rearrange-Array-Elements-by-Sign/rearrange_array_elements_by_sign.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> rearrangeArray(std::vector<int>& nums) {
        std::vector<int> positive = {};
        std::vector<int> negative = {};
        std::vector<int> output = {};

        for (auto n : nums) {
            if (n >= 0) {
                positive.push_back(n);
            }
            else {
                negative.push_back(n);
            }
        }

        for (int i = 0; i < positive.size(); i++) {
            output.push_back(positive[i]);
            output.push_back(negative[i]);
        }

        return output;
    }
};

int main() {
    std::vector<int> nums = {3,1,-2,-5,2,-4};

    Solution solution;

    std::vector<int> output = solution.rearrangeArray(nums);
    
    for (auto n : output){
        std::cout << n << " ";
    }
    
    return 0;
}